src/main/kotlin/app/services/impl/HttpKeyService.kt

```diff
@@ -23,28 +24,32 @@ import java.io.Reader
 import java.net.URLEncoder
 
 @Service
-class HttpKeyService(private val httpClientProvider: HttpClientProvider) : KeyService {
+class HttpKeyService(
+    private val httpClientProvider: HttpClientProvider,
+    private val uuidGenerator: UUIDGenerator) : KeyService {
 
     @Override
     @Retryable(value = [DataKeyServiceUnavailableException::class],
-            maxAttempts = maxAttempts,
-            backoff = Backoff(delay = initialBackoffMillis, multiplier = backoffMultiplier))
+        maxAttempts = maxAttempts,
+        backoff = Backoff(delay = initialBackoffMillis, multiplier = backoffMultiplier))
     @Throws(DataKeyServiceUnavailableException::class, DataKeyDecryptionException::class)
     override fun decryptKey(encryptionKeyId: String, encryptedKey: String): String {
-        logger.info("Decrypting encryptedKey: '$encryptedKey', keyEncryptionKeyId: '$encryptionKeyId'.")
+        val dksCorrelationId = uuidGenerator.randomUUID()
+        logger.info("Calling decryptKey: '$encryptedKey', keyEncryptionKeyId: '$encryptionKeyId', dks_correlation_id: '$dksCorrelationId'")
```

Comment:
I think this log line is less intuitive now? Maybe it should be updated to `Calling decryptKey: key:` so it's in line with the other log lines below

Reply:
Done!